Mage.Sets/src/mage/cards/i/IanMalcolmChaotician.java

```diff
@@ -0,0 +1,267 @@
+    }
+
+    private IanMalcolmChaoticianExileEffect(final IanMalcolmChaoticianExileEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public IanMalcolmChaoticianExileEffect copy() {
+        return new IanMalcolmChaoticianExileEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        UUID targetPlayerID = getTargetPointer().getFirst(game, source);
+        Player targetPlayer = game.getPlayer(targetPlayerID);
+        MageObject sourceObject = source.getSourceObject(game);
+        if (targetPlayer == null || sourceObject == null) {
+            return false;
+        }
+
+        Card card = targetPlayer.getLibrary().getFromTop(game);
+        if (card == null) {
+            return false;
+        }
+
+        UUID exileZoneId = CardUtil.getExileZoneId(game, sourceObject.getId(), sourceObject.getZoneChangeCounter(game));
+        targetPlayer.moveCardsToExile(card, source, game, true, exileZoneId, sourceObject.getIdName());
+        MageObjectReference sourceMOR = new MageObjectReference(source.getSourceId(), game);
+        IanMalcolmChaoticianWatcher.addCard(sourceMOR, card, game);
+        return true;
+    }
+}
+
+class IanMalcolmChaoticianCastEffect extends AsThoughEffectImpl {
+
+    IanMalcolmChaoticianCastEffect() {
+        super(AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, Duration.WhileOnBattlefield, Outcome.PlayForFree);
+        staticText = "During each player's turn, that player may cast a spell from among the cards they don't own " +
+                "exiled with {this}";
+    }
+
+    private IanMalcolmChaoticianCastEffect(final IanMalcolmChaoticianCastEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public IanMalcolmChaoticianCastEffect copy() {
+        return new IanMalcolmChaoticianCastEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (!game.isActivePlayer(affectedControllerId) || IanMalcolmChaoticianWatcher.checkUsed(source, game)) {
+            return false;
+        }
+        Card card = game.getCard(CardUtil.getMainCardId(game, sourceId));
+        if (card == null || card.isLand(game)){
+            return false;
+        }
+        MageObjectReference sourceMOR = new MageObjectReference(source.getSourceId(), game);
+        return !card.getOwnerId().equals(affectedControllerId)
+                && IanMalcolmChaoticianWatcher.checkExile(sourceMOR, card, game, 0);
+    }
+}
+
+class IanMalcolmChaoticianManaEffect extends AsThoughEffectImpl implements AsThoughManaEffect {
+
+    IanMalcolmChaoticianManaEffect() {
+        super(AsThoughEffectType.SPEND_OTHER_MANA, Duration.WhileOnBattlefield, Outcome.Benefit);
+        staticText = ", and mana of any type can be spent to cast it";
```

Comment:
is it possible to use some existing logic with CastManaAdjustment.AS_THOUGH_ANY_MANA_TYPE from #12040 ? or does this still need to be a separate effect for now? @Susucre might know more

Reply:
Yeah, i'm not terribly happy using duplicate code like this but the uses i see of the `CastManaAdjustment` enum use `CardUtil.makeCardPlayable`, which would defeat the custom `IanMalcolmChaoticianCastEffect`, or so i think. I
ll wait on a second optinion.